source/blog/2018-08-03-rspec-3-8-has-been-released.md

```diff
@@ -0,0 +1,112 @@
+dependency--that is, the suite only fails when the tests are run
+in a specific order.  `--bisect` will repeatedly run smaller and
+smaller subsets of your suite in order to narrow it down to the
+minimal set of specs needed to reproduce the failures.
+
+Since 3.3, this feature has been implemented by shelling out to
+the `rspec` command to run each subset. While simple and effective,
+we realized this approach was also quite inefficient. Each time the
+`rspec` command runs, it must boot RSpec and your application
+environment (which may include Rails and many other gems) from scratch.
+The cost of this can vary considerably from a couple hundred milliseconds
+to 30+ seconds on a large Rails app. In extreme cases, the runtime of
+`--bisect` can be dominated by the time it takes to boot your application
+environment over and over and over.
+
+In RSpec 3.8, we've improved bisect's performance by using forking
+on platforms that support it rather than shelling out. That way, we
+can boot your application environment _once_, and then fork a subprocess
+in which to run each subset of the test suite, avoiding the need to boot
+your application many times.
+
+The actual improvement you'll see in practice will vary widely, but
+in our [limited testing](https://github.com/rspec/rspec-core/pull/2511)
+it improved the runtime of `--bisect` by 33% in one case and an
+order-of-magnitude (108.9 seconds down to 11.7 seconds) in another.
+
+If you're looking to maximize the benefit of this change, you may
+want to pass some additional `--require` options when running a
+bisection in order to pre-load as much of your application environment
+as possible.
+
+### Core: Support the XDG base directory spec for configuration
+
+RSpec, like many command line tools, supports the use of options
+files, which can live at `.rspec` (for team project options)
+`~/.rspec` (for global personal options) or at `.rspec-local`
+(for personal project options -- this file should not be under
+source control). In RSpec 3.8, we've expanded this feature to
+support the [XDG Base Directory
+Specification](https://specifications.freedesktop.org/basedir-spec/latest/),
+which defines a standard way for tools to locate the global personal
+options file.  This gives users complete control over where this
+file is located rather than forcing it to live in their home directory.
+
+To use this feature, simply set the `$XDG_CONFIG_HOME` environment
+variable and put your RSpec options at `$XDG_CONFIG_HOME/rspec/options`.
+
+For more info, [read the spec](https://specifications.freedesktop.org/basedir-spec/latest/)
+or [check out the pull
+request](https://github.com/rspec/rspec-core/pull/2538).
+
+Thanks to Magnus Bergmark for implementing this feature!
+
+### Expectations: Formatted output length is now configurable
+
+TODO (Sam/Benoit): write this up since you authored this feature
+
+### Rails: ?
+
+TODO (Sam/Benoit/Jon): figure out what is notable
```

Comment:
```
### Rails:  `have_http_status` matcher upgraded to support Rails 5.2
Since Rails 5.2 the `have_http_status` matcher has issued deprecation warnings due to a change in the name
of the various states. In RSpec 3.8 these warnings are removed.

### Rails: View specs `stub_template` performance improved.

Thanks to Simon Coffey for implementing caching for `stub_template` that prevents unnecessary recreation of
templates. This improves performance by reducing the amount of allocation and setup performed.

### Rails: `rails_helper.rb`

Thank you to Koichi ITO and Alessandro Rodi for improving our generated `rails_helper.rb` with improved messages when migrations are pending, and bringing us in line with Rails standards.

### Rails: Scafold generator

Thanks to Laura Paakkinen, our scaffold generator now correctly respects longer namespaces for generating controllers / routes, e.g. `api/v1/posts` now generates `Api::V1::Posts`.
```

Reply:
Thanks, I've added these, except for the last one.  It's not clear what qualifies as a "longer namespace."  Is it one over 10 characters?  20?  30?  And also it doesn't seem that notable--it's just a bug fix, right?

Reply:
Its any multi part namespace over 2, I'm on the fence about it but similar to the other questionable one, it was more about highlighting contributor effort.

Reply:
Contributor effort has never been the basis on whether or not we highlight something as notable in our release blog post.  It's about the effect the change has for users of RSpec.

Reply:
Fair enough, how do you feel, given that our contributor count is quite low at the moment, about simply listing everyone who helped on this release at the end of the blog post? A "And thanks to <names> who helped with this release".

Reply:
All contributors will be listed in "Stats" section ([example](http://rspec.info/blog/2017/10/rspec-3-7-has-been-released/#stats)).  And they're also listed in the changelog for their individual contributions.  We don't need to list all the contributors 3 times in the same blog post.